tools/configure_server_settings.rb

```diff
@@ -2,21 +2,50 @@
 require File.expand_path("../config/environment", __dir__)
 require 'trollop'
 
+TYPES = %w(string integer boolean symbol float).freeze
+
 opts = Trollop.options(ARGV) do
   banner "USAGE:   #{__FILE__} -s <server id> -p <settings path separated by a /> -v <new value>\n" \
-         "Example: #{__FILE__} -d -s 1 -p reporting/history/keep_reports -v 42"
+         "Example (String): #{__FILE__} -s 1 -p reporting/history/keep_reports -v 42\n" \
+         "Example (Integer): #{__FILE__} -s 1 -p workers/worker_base/queue_worker_base/ems_metrics_collector_worker/defaults/count -v 1 -t integer\n" \
+         "Example (Boolean): #{__FILE__} -s 1 -p ui/mark_translated_strings -v true -t boolean\n" \
+         "Example (Symbol): #{__FILE__} -s 1 -p workers/worker_base/queue_worker_base/ems_metrics_collector_worker/defaults/poll_method -v escalate -t symbol\n" \
+         "Example (Float): #{__FILE__} -s 1 -p capacity/profile/1/vcpu_commitment_ratio -v 1.5 -t float"
 
-  opt :dry_run,  "Dry Run",                            :short => "d"
-  opt :serverid, "Server Id",                          :short => "s", :default => 0
-  opt :path,     "Path within advanced settings hash", :short => "p", :default => ""
-  opt :value,    "New value for setting",              :short => "v", :default => ""
+  opt :dry_run,  "Dry Run",                                  :short => "d"
+  opt :serverid, "Server Id",                                :short => "s", :type => :integer, :required => true
+  opt :path,     "Path within advanced settings hash",       :short => "p", :type => :string,  :required => true
+  opt :value,    "New Value for setting",                    :short => "v", :type => :string,  :required => true
+  opt :force,    "Force change value regardless of type",    :short => "f", :type => :boolean, :default  => false
+  opt :type,     "Type of value provided, #{TYPES.inspect}", :short => "t", :type => :string,  :default  => "string"
 end
 
 puts opts.inspect
 
 Trollop.die :serverid, "is required" unless opts[:serverid_given]
 Trollop.die :path,     "is required" unless opts[:path_given]
 Trollop.die :value,    "is required" unless opts[:value_given]
+Trollop.die :type,     "must be one of #{TYPES.inspect}" unless TYPES.include?(opts[:type])
+
+newval = ''
+
+# Grab the value that we have set and translate to appropriate var class
+case opts[:type]
+when "string"
+  newval = opts[:value]
+when "integer"
+  newval = opts[:value].to_i
+when "boolean"
+  if opts[:value].downcase == "true"
+    newval = true
+  elsif opts[:value].downcase == "false"
+    newval = false
+  end
+when "symbol"
+  newval = opts[:value].to_sym
+when "float"
+  newval = opts[:value].to_f
+end
```

Comment:
This would eliminate some of the extra assignments: (I hope rubocop doesn't hate me):

```ruby
newval =
  case opts[:type]
  when "integer"
    opts[:value].to_i
  when "boolean"
    if opts[:value].downcase == "true"
      true
    elsif opts[:value].downcase == "false"
      false
    end
  when "symbol"
    opts[:value].to_sym
  when "float"
    opts[:value].to_f
  end
```